src/core/ftd.cmake

```diff
@@ -41,6 +41,8 @@ target_include_directories(openthread-ftd PUBLIC ${OT_PUBLIC_INCLUDES} PRIVATE $
 target_sources(openthread-ftd PRIVATE ${COMMON_SOURCES})
 
 target_link_libraries(openthread-ftd
+    PUBLIC
```

Comment:
does these config have to be public?

Reply:
The alternative would be adding `PRIVATE` links to all the other targets that link against the corresponding `-ftd/-mtd/-radio/-rcp` libraries. Is that favorable? I can change the PR to that instead if so

Reply:
Yes, please. I think it's better not to forward flags if not necessary.